scripts/release.sh

```diff
@@ -95,6 +101,18 @@ echo "===== SETTINGS"
 echo "===== Fetch the remote content"
 git fetch ${REMOTE_NAME}
 
+if [ "${GENERATE_DOCS}" == "true" ]; then
+  echo "===== Create a new branch named [${STAGING_BRANCH}] (or switch to it if it already exists)"
+  git checkout -b ${STAGING_BRANCH} ${REMOTE_NAME}/${STAGING_BRANCH} || (git checkout ${STAGING_BRANCH} && git reset --hard ${REMOTE_NAME}/${STAGING_BRANCH})
+
+  echo "===== Generate the CRD schema documentation"
+  make gen-crd-doc
+  git commit -am "Auto-generated CRD schema documentation"
```

Comment:
I was expecting seeing only this chunk decorated with the conditional after line 127 to avoid the double checkout and the double push.

But this works, anyways. So if you like this, I'm not against.

Reply:
The double checkout is needed - because if we do this in the "middle" and it fails, we would have pushed changes to other branches without finishing the release - thus leaving our branches is a corrupted state. We need to do this first, finish the auto-generation - and only when successful, keep going to the release portion.

Reply:
This is what I mean - if we do the auto-gen after this, we will have pushed the new version branch, but then if auto-gen failed we won't have pushed the staging or current branch and things would get weird at that point.
```
echo "===== Create a new version branch named [${CURRENT_VERSION}] based on branch [${CURRENT_BRANCH}]"
git checkout -b ${CURRENT_VERSION} ${REMOTE_NAME}/${CURRENT_BRANCH}

echo "===== Set baseURL in config.toml for the branch [${CURRENT_VERSION}]"
sed -i "s/baseURL = .*/baseURL = \"https:\/\/${CURRENT_VERSION_WITH_DASHES}.kiali.io\"/" config.toml
git commit -am "Set base URL for branch: ${CURRENT_VERSION}"

echo "===== Push the new version branch [${CURRENT_VERSION}] to remote [${REMOTE_NAME}]"
git push ${REMOTE_NAME} ${CURRENT_VERSION}
```

Reply:
Well... there are several pushes here. I think a failure in almost any of them would require a manual fix.
But OK, let's leave this here to be more on the safe side.